bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gmos/Gmos.java

```diff
@@ -18,79 +14,136 @@
-        // transmits wavelengths that are a subset of the original range.
+        this.odp    = odp;
+        this.gp     = gp;
+
+        _detectorCcdIndex = detectorCcdIndex;
 
-        _observingStart = super.getStart();
-        _observingEnd = super.getEnd();
         _sampling = super.getSampling();
 
-        INSTR_PREFIX = INSTUMENT_PREFIX;
+        // TODO: filter is not yet defined, need to work with filter from gp, clean this up
+        if (!gp.getFilter().equals(GmosNorthType.FilterNorth.NONE) && !gp.getFilter().equals(GmosSouthType.FilterSouth.NONE)) {
+            _Filter = Filter.fromWLFile(getPrefix(), gp.getFilter().name(), getDirectory() + "/");
+            addFilter(_Filter);
+        }
+
+
+        FixedOptics _fixedOptics = new FixedOptics(getDirectory() + "/", getPrefix());
+        addComponent(_fixedOptics);
+
+
+        //Choose correct CCD QE curve
+        switch (gp.getCCDtype()) {
+            // E2V, site dependent
+            case E2V:
+                switch (gp.getSite()) {
+                    // E2V for GN: gmos_n_E2V4290DDmulti3.dat      => EEV DD array
+                    case GN:
+                        _detector = new Detector(getDirectory() + "/", getPrefix(), "E2V4290DDmulti3", "EEV DD array");
+                        _detector.setDetectorPixels(DETECTOR_PIXELS);
+                        if (detectorCcdIndex == 0) _instruments = new Gmos[]{this};
+                        break;
+                    // E2V for GS: gmos_n_cdd_red.dat              => EEV legacy
+                    case GS:
+                        _detector = new Detector(getDirectory() + "/", getPrefix(), "ccd_red", "EEV legacy array");
+                        _detector.setDetectorPixels(DETECTOR_PIXELS);
+                        if (detectorCcdIndex == 0) _instruments = new Gmos[]{this};
+                        break;
+                    default:
+                        throw new IllegalArgumentException();
+                }
+                break;
+            // Hamamatsu, both sites: gmos_n_CCD-{R,G,B}.dat        =>  Hamamatsu (R,G,B)
+            case HAMAMATSU:
+                String fileName = getCcdFiles()[detectorCcdIndex];
+                String name = getCcdNames()[detectorCcdIndex];
+                Color color = DETECTOR_CCD_COLORS[detectorCcdIndex];
+                _detector = new Detector(getDirectory() + "/", getPrefix(), fileName, "Hamamatsu array", name, color);
+                _detector.setDetectorPixels(DETECTOR_PIXELS);
+                if (detectorCcdIndex == 0)
+                    _instruments = createCcdArray();
+                break;
+            default:
+                throw new IllegalArgumentException();
+        }
+
+        if (detectorCcdIndex == 0) {
+            _dtv = new DetectorsTransmissionVisitor(gp.getSpectralBinning(),
+                    getDirectory() + "/" + getPrefix() + "ccdpix_red" + Instrument.getSuffix());
```

Comment:
Eventually these will be a `java.io.File`s rather than strings, I hope? Constructing paths with strings is error-prone (and technically you should use `File.separator` rather than `"/"` so it works on Windows).